src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -21,61 +47,689 @@ const ActorInit Door_Shutter_InitVars = {
+        DoorShutter_SetupAction(this, func_809976B8);
+        if (!(gSaveContext.eventChkInf[7] & 1)) { // not began Gohma fight
+            BossGoma* parent = (BossGoma*)this->dyna.actor.parent;
+            this->unk_164 = 10;
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+            func_8099803C(globalCtx, 2, 10, parent->subCameraId);
+            func_80033260(globalCtx, &this->dyna.actor, &this->dyna.actor.posRot.pos, 70.0f, 20, 8.0f, 500, 10, 1);
+        }
+    }
+}
+
+void func_809976B8(DoorShutter* this, GlobalContext* globalCtx) {
+    f32 mult;
+    if (this->unk_164 != 0) {
+        this->unk_164--;
+        mult = sinf((this->unk_164 * 250.0f) / 100.0f);
+        this->dyna.actor.shape.unk_08 = ((this->unk_164 * 3.0f) / 10.0f) * mult;
+    }
+}
+
+void func_80997744(DoorShutter* this, GlobalContext* globalCtx) {
+    f32 phi_f0;
+
+    osSyncPrintf("FHG SAKU START !!\n");
+    if (this->unk_164 != 0) {
+        this->unk_164--;
+    }
+    phi_f0 = (this->unk_164 & 1) ? -3.0f : 0.0f;
+    Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, -34.0f + phi_f0, 1.0f, 20.0f, 0.0f);
+    osSyncPrintf("FHG SAKU END !!\n");
+}
+
+void DoorShutter_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DoorShutter* this = THIS;
+    Player* player = PLAYER;
+
+    if (!((s32)player->stateFlags1 & 0x100004C0) || (this->actionFunc == DoorShutter_SetupType)) {
+        this->actionFunc(this, globalCtx);
+    }
+}
+
+Gfx* func_80997838(GlobalContext* globalCtx, DoorShutter* this, Gfx* p) {
+    MtxF mtx;
+    f32 angle = 0.0f;
+    f32 yScale = this->unk_166 * 0.01f;
+    s32 i;
+
+    Matrix_Get(&mtx);
+    for (i = 0; i != ARRAY_COUNT(D_809982B4); i++) {
+        Matrix_RotateZ(angle, MTXMODE_APPLY);
+        if (!(i & 1)) {
+            Matrix_Translate(0.0f, 800.0f, 0.0f, MTXMODE_APPLY);
+        } else if (i == 1 || i == 7) {
+            Matrix_Translate(0.0f, 848.52f, 0.0f, MTXMODE_APPLY);
+        } else {
+            Matrix_Translate(0.0f, 989.94f, 0.0f, MTXMODE_APPLY);
+        }
+        if (this->unk_166 != 100) {
+            Matrix_Scale(1.0f, yScale, 1.0f, MTXMODE_APPLY);
+        }
+        gSPMatrix(p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_door_shutter.c", 1991),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(p++, D_809982B4[i]);
+        angle -= M_PI / 4;
+        Matrix_Put(&mtx);
+    }
+    return p;
+}
+
+s32 func_80997A34(DoorShutter* this, GlobalContext* globalCtx) {
+    s32 phi_a1;
+    s32 phi_a0;
+
+    if (Player_InCsMode(globalCtx) != 0) {
+        return true;
+    }
+    phi_a0 = (s16)(func_8002DAC0(&this->dyna.actor, &globalCtx->view.eye) - this->dyna.actor.shape.rot.y);
+    phi_a1 = (s16)(this->dyna.actor.yawTowardsLink - this->dyna.actor.shape.rot.y);
+    phi_a0 = ABS(phi_a0);
+    phi_a1 = ABS(phi_a1);
+    if (!(phi_a1 >= 0x4000 || phi_a0 <= 0x4000) || (phi_a1 >= 0x4001 && phi_a0 < 0x4000)) {
```

Comment:
```suggestion
    if (!(phi_a1 >= 0x4000 || phi_a0 <= 0x4000) || (phi_a1 > 0x4000 && phi_a0 < 0x4000)) {
```
can this conditional be written like:
```suggestion
    if ((phi_a1 < 0x4000 && phi_a0 > 0x4000) || (phi_a1 > 0x4000 && phi_a0 < 0x4000)) {
```
more readable imo

Reply:
Worked